go/mysqlconn/replication/binlog_event_common.go

```diff
@@ -528,66 +528,81 @@ func cellLength(data []byte, pos int, typ byte, metadata uint16) (int, error) {
 	}
 }
 
-// cellData returns the data for a cell as a string. It only uses the
-// querypb.Type value for the signed flag.
-func cellData(data []byte, pos int, typ byte, metadata uint16, styp querypb.Type) (string, int, error) {
+// CellValue returns the data for a cell as a sqltypes.Value, and how
+// many bytes it takes. It only uses the querypb.Type value for the
+// signed flag.
+func CellValue(data []byte, pos int, typ byte, metadata uint16, styp querypb.Type) (sqltypes.Value, int, error) {
 	switch typ {
 	case TypeTiny:
 		if sqltypes.IsSigned(styp) {
-			return strconv.FormatInt(int64(int8(data[pos])), 10), 1, nil
+			return sqltypes.MakeTrusted(querypb.Type_INT8,
```

Comment:
You could use strconv.AppendInt instead, like sqltypes.BuildValue does. I'd almost recommend using that function instead, but then you'd have to add unnecessary error-check.